app/src/main/kotlin/com/wire/android/ui/main/navigation/MainTopBar.kt

```diff
@@ -0,0 +1,116 @@
+package com.wire.android.ui.main.navigation
+
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material.DrawerValue
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.ScaffoldState
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Menu
+import androidx.compose.material.rememberDrawerState
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.navigation.NavController
+import androidx.navigation.compose.rememberNavController
+import com.wire.android.R
+import com.wire.android.ui.common.SearchBarCollapsed
+import com.wire.android.ui.common.UserProfileAvatar
+import com.wire.android.ui.theme.WireLightColors
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@ExperimentalAnimationApi
+@ExperimentalComposeUiApi
+@Composable
+fun MainTopBar(scope: CoroutineScope, scaffoldState: ScaffoldState, navController: NavController, hasSearchBar: Boolean = false) {
```

Comment:
Maybe Detekt will also complain about this method being too long.

If it was any other actual logic I would agree. But being this a UI, which could be seen as analog to XML files, this isn't bad.

Maybe yes, we could split into smaller chunks, like `@Composable fun TopBarContent()` ?

Or just disable these checks for things annotated with `Composable`. In `config/detekt/detekt.yml`, line 84:

```
    ignoreAnnotated: ['Composable']
```

Reply:
I think I will add the Composable objects to this rule as usually these components tend to have long signature method headers